DM: support for MySQL 8.0 GA (#15959)
pingcap/docs: dm/dm-overview.md

    - DM does not support migrating `charset=GBK` tables to TiDB clusters earlier than v5.4.0.

+ Binlog compatibility

>     - DM does not support the MySQL 8.0 new feature binlog [Transaction_payload_event](https://dev.mysql.com/doc/dev/mysql-server/8.1.0/classbinary__log_1_1Transaction__payload__event.html). Using binlog Transaction_payload_event may result in data inconsistency of upstream and downstream.

Author: ```suggestion
    - DM does not support the MySQL 8.0 new feature binlog [Transaction_payload_event](https://dev.mysql.com/doc/dev/mysql-server/8.1.0/classbinary__log_1_1Transaction__payload__event.html). Using binlog Transaction_payload_event might result in data inconsistency between upstream and downstream.
```

>     - DM does not support the MySQL 8.0 new feature binlog [Transaction_payload_event](https://dev.mysql.com/doc/dev/mysql-server/8.1.0/classbinary__log_1_1Transaction__payload__event.html). Using binlog Transaction_payload_event might result in data inconsistency between upstream and downstream.

Author: ```suggestion
    - DM does not support the MySQL 8.0 new feature binlog [Transaction_payload_event](https://dev.mysql.com/doc/refman/8.0/en/binary-log-transaction-compression.html). Using binlog Transaction_payload_event might result in data inconsistency between upstream and downstream.
```
